### ISSUE TITLE
Operations |=, &=, ^= 

### ISSUE THREAD

**Issue by eadf** [softdevteam/vob#61]
Do you want a [PR](https://github.com/eadf/vob/commit/87c879dae1e58fa6385359b190b3d1ab3a68ac8e) for the std::ops::Bit*Assign operations?

**Comment by eadf** [comment]
BitAnd (and friends) can take any combinations of (&self,self) in total four impls. But the variants that consumes rhs are not as useful imho. [Here](https://github.com/eadf/vob/commit/015fb5b5fabd5e9991cd17aed39ff433c83450ef) are the (&self,&self) and (self,&self) implementations. 

And yes, the variants that consumes lhs uses mutate in place.

About the panic: There is only one way to avoid this as there are no TryBitAndAssign traits. That would be to pad the shorter Vob. But with what? True or False? And shouldn't this behavior be carried over to and(), or() and xor() as well?

As you can see these impls are kind of "wordy", maybe it is time to break out the tests into separate module, making the file size manageable again?

**Comment by ltratt** [comment]
Devil's advocate (I haven't made my mind up!): we *could* still panic on unequal length `Vob`s in the same way that `ops::Div` implementations will tend to panic if `rhs` is 0.

**Comment by ltratt** [comment]
Having thought about this more, I think that you've hit the nail on the head by asking "[means we have to] pad the shorter Vob. But with what? True or False?" Whatever we choose, I think the answer is surprising in some cases, and more likely to cause hard-to-debug bugs than would a `panic`.

I'm therefore inclined to say that #62 shouldn't pad any of these, and we should `panic` on unequal `Vob`s. As a bonus, if one day we work out some clever mechanism for padding, it's easier to go from "panic on unequal length" to "sensible-padding on equal length" than the other way around.

**Comment by eadf** [comment]
Fair enough, it's easy to rollback the padding stuff ([result will look like this](https://github.com/eadf/vob/commit/015fb5b5fabd5e9991cd17aed39ff433c83450ef)). Just give the final word and I'll do it. 

There is a clippy warning: "this `impl` can be derived" and one deprecation notification: "usize::max_value()".
Want me to fix those with this PR?

The semver documentation is a bit vague about implementing traits. But I think this is a minor update (3.0.1)
I'm downloading the [rust-semverver](https://github.com/rust-lang/rust-semverver) tool right now.

**Comment by ltratt** [comment]
Good idea!

 * fix the clippy issue? *Yes* (in a separate commit)
 * fix the deprecation issue? *Yes* (in a separate commit)
 * make file-size manageable by moving tests to a separate module? *No* (maybe a second PR?)
 * implement all permutations of [&Self,Self] for the operations? *I'm not sure what this means*
 * let the operations panic? *Yes*
